tpm2_pytss/tsskey.py

```diff
@@ -0,0 +1,82 @@
+"""
+SPDX-License-Identifier: BSD-3
+"""
+
+from ._libtpm2_pytss import lib
+from .types import *
+from asn1crypto.core import ObjectIdentifier
+
+
+parent_rsa_template = TPMT_PUBLIC() # FIXME, use template from tpm2-openssl
+parent_ecc_template = TPMT_PUBLIC() # FIXME, use template from tpm2-openssl
+
+loadablekey_oid = ObjectIdentifier("2.23.133.10.1.3")
+
+class TSSPrivKey(object):
+    def __init__(self, private, public, emptyAuth=True, parent=lib.TPM2_RH_OWNER):
+        self.private = private
+        self.public = public
+        self.emptyAuth = emptyAuth
+        self.parent = parent
+
+    @property
+    def private(self):
+        return self.private
+
+    @property
+    def public(self):
+        return self.public
+
+    @property
+    def emptyAuth(self):
+        if self.emptyAuth:
+            return True
+        return False
+
+    @property
+    def parent(self):
+        return self.parent
+        
+    def toDER(self):
+        # asn1 here
+        pass
+        
+    def toPEM(self):
+        # serialize here
+        der = self.toDER()
+
+    def _getparent(self, ectx):
+        if self.parent == lib.TPM2_RH_OWNER and self.public.publicArea.type == TPM2_ALG.RSA:
+            template = parent_rsa_template
+        elif self.parent == lib.TPM2_RH_OWNER and self.public.publicArea.type == TPM2_ALG.ECC:
+            template = parent_ecc_template
+        else:
+            return ectx.TR_GetTpmPublic(self.parent)
+        # generate parent here
+
+    def load(self, ectx, password=None):
+        if not password and self.emptyAuth:
+            raise Exception("no password specified but it is required")
```

Comment:
I think the check is wrong, because if password is not set (ie a string with len 0 or None) and it is the emptyAuth,  that's correct. Ie it is an emptyAuth.

```python
if password and self.emptyAuth:
   raise RuntimeError('Password specified but it has an empty auth value')
```
or
```python
if not password and not self.emptyAuth:
   raise RuntimeError('No password specified but it is required')
```

Reply:
yep, it's wrong, I'll fix
the question if we should throw a warning if password is set, but emptyAuth is true, it would be good feedback

Reply:
Warning is fine IMO.